### PR TITLE
TMDM-14007 No Foreign Key picker (FK Picker) in the Search Bar in MDM Web UI

### PR DESCRIPTION
**Pull request by zfkang-talend**
JIRA: https://jira.talendforge.org/browse/TMDM-14007

**What is the current behavior?** (You should also link to an open issue here)
In search panel of the WEB UI, the FK field less the FK Picker, only the text input.

**What is the new behavior?**
add the FK picker but the drop-down list changed to text input field.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java [hunk 1]
```diff
@@ -406,6 +414,6 @@ public String getOriginForeignKeyFilter() {
     }
 
     public boolean isSearch() {
```
Comment (lpw82gf) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java]: Why not rename to `isFromSearchPanel`

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java [hunk 2]
```diff
@@ -80,6 +84,30 @@ public ForeignKeyField(TypeModel dataType) {
...
         this.setFireChangeEventOnSetValue(true);
+        this.isFromSearchPanel = false;
+        String[] foreignKeyPathArray = foreignKeyPath.split("/"); //$NON-NLS-1$
+        if (foreignKeyPathArray.length > 0) {
+            foreignConceptName = foreignKeyPathArray[0];
+        }
+    }
+
+    public ForeignKeyField(TypeModel dataType, boolean isFromSearchPanel) {
+        // if isFromSearchPanel=true, suggestBox don't initialize and initialize the textField
```
Comment (lpw82gf) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java]: `// if isFromSearchPanel=true, won't initialize suggestBox but will initialize the textField`

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java [hunk 3]
```diff
@@ -70,6 +72,8 @@
 
     protected String originForeignKeyFilter;
 
+    protected boolean isFromSearchPanel; // if create FK field from search panel, isSearch=true
```
Comment (lpw82gf) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java]: ` //Check if create FK field from search panel`

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/inputfield/creator/SearchFieldCreator.java [hunk 4]
```diff
@@ -97,6 +94,13 @@ private static Field createForeignKeyField(TypeModel typeModel) {
         return fkField;
     }
 
+    public static Field createForeignKeyFieldForInput(TypeModel typeModel) {
```
Comment (lpw82gf) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/inputfield/creator/SearchFieldCreator.java]: Could refactor to have one method like `private static Field createForeignKeyField(TypeModel typeModel, boolean isTextInput)`

File: org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java [hunk 5]
```diff
@@ -206,11 +255,24 @@ public void setValue(ForeignKeyBean foreignKeyBean) {
...
                 suggestBox.setValue(foreignKeyBean);
             }
         }
+        if (withTextInput) {
+            if (foreignKeyBean != null) {
+                String value = foreignKeyBean.getId();
+                // the value return by FK Picker wrapped by '[' and ']'
+                // in Advance Search, the value from the input, only the FK id.
+                // if the value wrapped by '[' and ']', will remove them
+                if (value.startsWith("[") && value.endsWith("]")) {
```
Comment (hw-zhu) [inline org.talend.mdm.webapp.browserecords/src/main/java/org/talend/mdm/webapp/browserecords/client/widget/ForeignKey/ForeignKeyField.java]: $NON-NLS-1$

**Pull request merged**